src/content/docs/infrastructure/host-integrations/host-integrations-list/temporal-monitoring-integration.mdx

```diff
@@ -64,66 +68,131 @@ You can check the Temporal server running on the below urls:
+
+public class MetricsWorker {
+    static final String WORK_FLOW_TASK_QUEUE = "WORK_FLOW_TASK_QUEUE";//This can be a work flow task name used to differentiate the metrics logs from other work flow
+
+    public static void main(String[] args) {
+
+        PrometheusMeterRegistry registry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+        StatsReporter reporter = new MicrometerClientStatsReporter(registry);
+
+        // set up a new scope, report every 10 seconds
+        Scope scope = new RootScopeBuilder()
+                .tags(ImmutableMap.of(
+                        "workerCustomTag1",
+                        "workerCustomTag1Value",
+                        "workerCustomTag2",
+                        "workerCustomTag2Value"))
+                .reporter(reporter)
+                .reportEvery(com.uber.m3.util.Duration.ofSeconds(10));
+
+        // For Prometheus collection, expose the scrape endpoint at port 8077. See Micrometer documentation for details on starting the Prometheus scrape endpoint. For example,
+        HttpServer scrapeEndpoint = MetricsUtils.startPrometheusScrapeEndpoint(registry, 8077); //note: MetricsUtils is a utility file with the scrape endpoint configuration. See Micrometer docs for details on this configuration.
+        // Stopping the starter stops the HTTP server that exposes the scrape endpoint.
+        //Runtime.getRuntime().addShutdownHook(new Thread(() -> scrapeEndpoint.stop(1)));
+
+        //Create Workflow service stubs to connect to the Frontend Service.
+        WorkflowServiceStubs service = WorkflowServiceStubs.newServiceStubs(
+                WorkflowServiceStubsOptions.newBuilder()
+                        .setMetricsScope(scope) //set the metrics scope for the WorkflowServiceStubs
+                        .build());
+
+        //Create a Workflow service client, which can be used to start, signal, and query Workflow Executions.
+        WorkflowClient yourClient = WorkflowClient.newInstance(service,
+                WorkflowClientOptions.newBuilder().build());
+
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> scrapeEndpoint.stop(1)));
+        // Add metrics scope to workflow service stub options
+        WorkerFactory factory = WorkerFactory.newInstance(yourClient);
+
+        Worker worker = factory.newWorker(WORK_FLOW_TASK_QUEUE);
+        worker.registerWorkflowImplementationTypes(SampleWorkflowImpl.class);//Design a workflow incorporating temporal elements and invoking activities within it. Determine where to capture metrics logs and register them with the worker
+        worker.registerActivitiesImplementations(new SampleActivityImpl());  // Develop an Activity interface utilizing temporal annotations, proceed to its implementation, and establish a connection with the worker by mapping it to registerActivities
+
+        factory.start();
+    }
+}
+```
+2. MetricsUtils is a utility file with the scrape endpoint configuration.
```

Comment:
```suggestion
2. `MetricsUtils` is a utility file with the scrape endpoint configuration.
```

Reply:
Hi, @rhetoric101 , 
Thank you for your feedback. I've addressed your comments. Kindly review it.